### DISCUSSION TITLE
Curiosity: why determinism can't be assured?

### DISCUSSION THREAD

**Discussion by farfalk** [godot-jolt/godot-jolt#557] [General]
First, thank you for the incredible work. 

Second, as per title. Is it a limitation of the GDExtension framework or of the Godot Engine as a whole?

If this discussion is useless or out of topic for the repo, I'm sorry: feel free to delete it with no warning.

Cheers!

**Comment by mihe** [comment]
See #301 and #548.

tl;dr: The physics engine is only deterministic as long as the input is, and since Godot (which doesn't guarantee determinism) is what steps the simulation and provides the input to it (one way or another) I struggle to think of how I could make those kinds of guarantees. If Godot itself ever starts making those kinds of guarantees then maybe there's some discussion to be had.

**Reply by farfalk** [reply]
My bad, I didn't notice the "is: open" filter in the discussion search and so didn't read the "Quick Question about determinism in Godot Jolt" discussion.

So, even more tl;dr: as in everything, garbage in = garbage out, and there's no guarantee that Godot is giving good stuff or garbage as input to the Jolt system.

Thank you for your response and your patience!
